applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/prebuckling_strategy.hpp

```diff
@@ -0,0 +1,980 @@
+
+    typedef typename TDenseSpace::MatrixType DenseMatrixType;
+
+    typedef TSparseSpace SparseSpaceType;
+
+    typedef typename TSparseSpace::VectorPointerType SparseVectorPointerType;
+
+    typedef typename TSparseSpace::MatrixPointerType SparseMatrixPointerType;
+
+    typedef typename TSparseSpace::MatrixType SparseMatrixType;
+
+    typedef typename TSparseSpace::VectorType SparseVectorType;
+
+    typedef ConvergenceCriteria<TSparseSpace, TDenseSpace> ConvergenceCriteriaType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor.
+    PrebucklingStrategy(
+        ModelPart &rModelPart,
+        SchemePointerType pScheme,
+        BuilderAndSolverPointerType pEigenSolver,
+        BuilderAndSolverPointerType pBuilderAndSolver,
+        typename ConvergenceCriteriaType::Pointer pConvergenceCriteria,
+        int MaxIteration,
+        double InitialStep,
+        double SmallStep,
+        double BigStep,
+        double ConvergenceRatio )
+        : SolvingStrategy<TSparseSpace, TDenseSpace, TLinearSolver>(rModelPart)
+    {
+        KRATOS_TRY
+
+        mpScheme = pScheme;
+
+        mpEigenSolver = pEigenSolver;
+
+        mpBuilderAndSolver = pBuilderAndSolver;
+
+        mpConvergenceCriteria = pConvergenceCriteria;
+
+        mMaxIteration = MaxIteration;
+
+        mInitialStep = InitialStep;
+
+        mSmallStep = SmallStep;
+
+        mBigStep = BigStep;
+
+        mConvergenceRatio = ConvergenceRatio; 
+
+        // Set Eigensolver flags
+        mpEigenSolver->SetDofSetIsInitializedFlag(false);
+        mpEigenSolver->SetReshapeMatrixFlag(false);
+        mpEigenSolver->SetCalculateReactionsFlag(false);
+        // Set Static Builder and Solver flags
+        mpBuilderAndSolver->SetDofSetIsInitializedFlag(false);
+        mpBuilderAndSolver->SetCalculateReactionsFlag(false);
+        mpBuilderAndSolver->SetReshapeMatrixFlag(false);
+
+        // Set EchoLevel to the default value (only time is displayed)
+        this->SetEchoLevel(1);
+
+        // default rebuild level (build at each solution step)
+        this->SetRebuildLevel(1);
+
+        //Set Matrices and Vectors to empty pointers
+        mpStiffnessMatrix = TSparseSpace::CreateEmptyMatrixPointer();
+        mpStiffnessMatrixPrevious = TSparseSpace::CreateEmptyMatrixPointer();
+        mpDx = TSparseSpace::CreateEmptyVectorPointer();
+        mpRHS = TSparseSpace::CreateEmptyVectorPointer();
+
+        KRATOS_CATCH("")
+    }
+
+    /// Deleted copy constructor.
+    PrebucklingStrategy(const PrebucklingStrategy &Other) = delete;
+
+    /// Destructor.
+    ~PrebucklingStrategy() override
+    {
+        // Clear() controls order of deallocation to avoid invalid memory access
+        // in some special cases.
+        // warning: BaseType::GetModelPart() may be invalid here.
+        this->Clear();
+    }
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void SetIsInitialized(bool val)
```

Comment:
Add Doxygen description to everything please

Reply:
Added Doxygen Descriptions.